### PR TITLE
Create wrappers for web elements

### PR DESCRIPTION
**Pull request by Astashevich**

### CODE REVIEW

File: src/main/java/constants/OnlinerConstants.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package by.onliner.constants;
+package constants;
```
Comment (Anassstasi) [inline src/main/java/constants/OnlinerConstants.java]: Why have you updated this?

Comment (Astashevich) [reply src/main/java/constants/OnlinerConstants.java]: Because i have problems with idea, and i want to update my packages

File: src/main/java/elements/widget/ButtonImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    public void click() {
+        getWrappedElement().click();
+    }
+
+    /***
+     * Checks the button clickable on a page
+     * @return boolean true for a clickable Button, or false if passed a Element Not Interactable Exception.
+     */
+    @Override
+    public boolean isClickable() {
```
Comment (Anassstasi) [inline src/main/java/elements/widget/ButtonImpl.java]: Please review this method, check for implementation in google. Make it more efficient

File: src/main/java/elements/widget/EntryField.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package elements.widget;
+
+import elements.base.Element;
+import elements.base.ImplementedBy;
+
+@ImplementedBy(EntryFieldImpl.class)
+public interface EntryField extends Element {
```
Comment (Anassstasi) [inline src/main/java/elements/widget/EntryField.java]: What is entryField? You mean textField?

File: src/main/java/page/components/Menu.java [hunk 4]
```diff
@@ -30,11 +31,6 @@ public void waitForComponentOpened() {
      */
     public void openShoppingCartPage() {
         waitForComponentOpened();
-//        if (!shoppingCartButton.isEnabled()) {
```
Comment (Anassstasi) [inline src/main/java/page/components/Menu.java]: Why have you removed this?Removing is not part of this PR

File: src/main/java/page/components/ShoppingCartPopup.java [hunk 5]
```diff
@@ -31,6 +31,7 @@ public boolean isShoppingCartPopupVisible() {
      * Click on the popup close button.
      */
     public void closeShoppingCartPopup() {
+        waitForComponentOpened();
```
Comment (Anassstasi) [inline src/main/java/page/components/ShoppingCartPopup.java]: not part of this PR

File: src/main/java/page/pages/CatalogItemPage.java [hunk 6]
```diff
@@ -44,8 +44,8 @@ public void addToCart() {
         waitForPageOpened();
         addToShoppingCartButton.click();
         Waiter.waitForVisibility(greenShoppingCartButton);
-//        if (shoppingCartPopup.isShoppingCartPopupVisible()) {
-//            shoppingCartPopup.closeShoppingCartPopup();
-//        }
+        /*if (shoppingCartPopup.getShoppingCartPopupCloseButton().isDisplayed()) {
```
Comment (Anassstasi) [inline src/main/java/page/pages/CatalogItemPage.java]: These changes not part of this PR

File: src/main/java/page/pages/ShoppingCartPage.java [hunk 7]
```diff
@@ -1,28 +1,26 @@
...
 
     @FindBy(xpath = "//div[contains(@class, 'cart-form__title')]")
-    private WebElement openedShoppingCartMessage;
+    private Text openedShoppingCartMessage;
 
     @FindBy(className = "cart-form__control")
-    private WebElement removeFromCartButton;
-
-    @FindBy(xpath = "//div[contains(@class, 'cart-form__offers-unit_primary')]")
-    private WebElement cartUnitForm;
```
Comment (Anassstasi) [inline src/main/java/page/pages/ShoppingCartPage.java]: Why is this element removed in this PR?

File: src/test/java/by/onliner/BaseTest.java [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
 import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
-import static by.onliner.driver.DriverManager.getDriver;
+import page.pages.CatalogItemPage;
+import page.pages.MainPage;
+import page.pages.ShoppingCartPage;
+import static driver.DriverManager.getDriver;
 
-public abstract class AbstractTest {
+public abstract class BaseTest {
```
Comment (Anassstasi) [inline src/test/java/by/onliner/BaseTest.java]: Why have you renamed it?

File: src/main/java/constants/OnlinerConstants.java (hunk 1 above)
Comment (Astashevich) [reply src/main/java/constants/OnlinerConstants.java]: Fixed

File: src/main/java/elements/widget/ButtonImpl.java (hunk 2 above)
Comment (Astashevich) [reply src/main/java/elements/widget/ButtonImpl.java]: Method deleted

File: src/main/java/elements/widget/EntryField.java (hunk 3 above)
Comment (Astashevich) [reply src/main/java/elements/widget/EntryField.java]: Rename to TextInput

File: src/main/java/page/components/ShoppingCartPopup.java (hunk 5 above)
Comment (Astashevich) [reply src/main/java/page/components/ShoppingCartPopup.java]: Deleted

File: src/test/java/by/onliner/BaseTest.java (hunk 8 above)
Comment (Astashevich) [reply src/test/java/by/onliner/BaseTest.java]: Rollback

File: src/main/java/by/onliner/page/AbstractPage.java [hunk 9]
```diff
@@ -1,20 +1,21 @@
...
 
     public AbstractPage() {
         driver = DriverManager.getDriver();
-        PageFactory.initElements(this.driver, this);
+        ElementFactory.initElements(this.driver, this);
     }
 
     /***
-     * Checks the element presence on a page.
+     * Checks the element presence on a by.onliner.page.
```
Comment (Anassstasi) [inline src/main/java/by/onliner/page/AbstractPage.java]: by.onliner.page ?

File: src/test/java/by/onliner/AbstractTest.java [hunk 10]
```diff
@@ -16,7 +16,7 @@
     protected ShoppingCartPage shoppingCartPage;
 
     /***
-     * Set and open the local thread browser type with page initialisation.
+     * Set and open the local thread browser type with by.onliner.page initialisation.
```
Comment (Anassstasi) [inline src/test/java/by/onliner/AbstractTest.java]: Please check all the java docs :)

File: src/main/java/by/onliner/page/AbstractPage.java (hunk 9 above)
Comment (Astashevich) [reply src/main/java/by/onliner/page/AbstractPage.java:18]: Yes, it was been a page of all time

Comment (Astashevich) [reply src/main/java/by/onliner/page/AbstractPage.java]: Fixed

**Review by Anassstasi — APPROVED** [review]
LGTM!
